src/extensions/SubmittedFormExtension.php

```diff
@@ -23,12 +23,14 @@ public function updateAfterProcess()
     {
         // cleanup partial submissions
         $partialID = Controller::curr()->getRequest()->getSession()->get(PartialSubmissionController::SESSION_KEY);
-        /** @var PartialFormSubmission $partialForm */
-        $partialForm = PartialFormSubmission::get()->byID($partialID);
-        if ($partialForm) {
-            $partialForm->delete();
-            $partialForm->destroy();
+        if ($partialID !== null) {
+            /** @var PartialFormSubmission $partialForm */
+            $partialForm = PartialFormSubmission::get()->byID($partialID);
+            if ($partialForm) {
```

Comment:
These two ifs can be combined in to one:
`if ($partialID !== null && ($partialForm = $partialFormSubmission::get()->byID($partialID))) {`

It would save a nested if.
Or, maybe even simpler, 
`($partialForm = $partialFormSubmission::get()->byID((int)$partialID))`
If the partialID is `null`, it would become 0, and thus not finding anything. If the partialID doesn't exist, it would still work.

Reply:
Well, I want to avoid conditional assignment as much as I can.

Reply:
```
$partialID = Controller::curr()->getRequest()->getSession()->get(PartialSubmissionController::SESSION_KEY);
if ($partialID == null) {
    return;
}

/** @var PartialFormSubmission $partialForm */
$partialForm = PartialFormSubmission::get()->byID($partialID);
if ($partialForm === null) {
    return;
}

$partialForm->delete();
$partialForm->destroy();
Controller::curr()->getRequest()->getSession()->clear(PartialSubmissionController::SESSION_KEY);
```

Reply:
I prefer the last idea, I like early returns and verbosity :)

Reply:
@Firesphere ready for PR